cub/cub/device/dispatch/dispatch_transform.cuh

```diff
@@ -0,0 +1,1128 @@
+struct PoorExpected
+{
+  alignas(T) char storage[sizeof(T)];
+  cudaError_t error;
+
+  _CCCL_HOST_DEVICE PoorExpected(T value)
+      : error(cudaSuccess)
+  {
+    new (storage) T(::cuda::std::move(value));
+  }
+
+  _CCCL_HOST_DEVICE PoorExpected(cudaError_t error)
+      : error(error)
+  {}
+
+  _CCCL_HOST_DEVICE explicit operator bool() const
+  {
+    return error == cudaSuccess;
+  }
+
+  _CCCL_HOST_DEVICE T& operator*()
+  {
+    _CCCL_DIAG_PUSH
+    _CCCL_DIAG_SUPPRESS_GCC("-Wstrict-aliasing")
+    return reinterpret_cast<T&>(storage);
+    _CCCL_DIAG_POP
+  }
+
+  _CCCL_HOST_DEVICE const T& operator*() const
+  {
+    _CCCL_DIAG_PUSH
+    _CCCL_DIAG_SUPPRESS_GCC("-Wstrict-aliasing")
+    return reinterpret_cast<const T&>(storage);
+    _CCCL_DIAG_POP
+  }
+};
+
+_CCCL_HOST_DEVICE inline PoorExpected<int> get_max_shared_memory()
+{
+  // gevtushenko promised me that I can assume that stream belongs to the currently active device
+  int device = 0;
+  auto error = CubDebug(cudaGetDevice(&device));
+  if (error != cudaSuccess)
+  {
+    return error;
+  }
+
+  int max_smem = 0;
+  error        = CubDebug(cudaDeviceGetAttribute(&max_smem, cudaDevAttrMaxSharedMemoryPerBlock, device));
+  if (error != cudaSuccess)
+  {
+    return error;
+  }
+
+  return max_smem;
+}
+
+template <bool RequiresStableAddress,
+          typename Offset,
+          typename RandomAccessIteratorTupleIn,
+          typename RandomAccessIteratorOut,
+          typename TransformOp,
+          typename PolicyHub = policy_hub<RequiresStableAddress, RandomAccessIteratorTupleIn>>
+struct dispatch_t;
+
+template <bool RequiresStableAddress,
+          typename Offset,
+          typename... RandomAccessIteratorsIn,
+          typename RandomAccessIteratorOut,
+          typename TransformOp,
+          typename PolicyHub>
+struct dispatch_t<RequiresStableAddress,
+                  Offset,
+                  ::cuda::std::tuple<RandomAccessIteratorsIn...>,
+                  RandomAccessIteratorOut,
+                  TransformOp,
+                  PolicyHub>
+{
+  static_assert(::cuda::std::is_same<Offset, ::cuda::std::int32_t>::value
+                  || ::cuda::std::is_same<Offset, ::cuda::std::int64_t>::value,
+                "cub::DeviceTransform is only tested and tuned for 32-bit or 64-bit signed offset types");
+
+  ::cuda::std::tuple<RandomAccessIteratorsIn...> in;
+  RandomAccessIteratorOut out;
+  Offset num_items;
+  TransformOp op;
+  cudaStream_t stream;
+
+  using kernel_ptr_t =
+    decltype(&transform_kernel<typename PolicyHub::max_policy,
+                               Offset,
+                               TransformOp,
+                               RandomAccessIteratorOut,
+                               THRUST_NS_QUALIFIER::try_unwrap_contiguous_iterator_t<RandomAccessIteratorsIn>...>);
+  kernel_ptr_t kernel;
+
+  static constexpr int loaded_bytes_per_iter =
+    static_cast<int>(loaded_bytes_per_iteration<RandomAccessIteratorsIn...>());
```

Comment:
minor hint: uniform initialization `int{}` for compile-time constants

Reply:
We still build C++11 so not possible

Reply:
I changed the signature of `loaded_bytes_per_iteration` to return `int` now, since that's the type that is needed in all locations. Do you still insist on direct-initialization?